File: Difficulty 3.0/Wood Cutting/WoodCutting.java
import java.util.Scanner;
import java.util.ArrayList;
import java.util.Collections;

public class WoodCutting {
    public static void main (String [] args) {
        Scanner sc = new Scanner(System.in);

        int test = sc.nextInt();

        for (int i = 0; i < test; i++) {
            int people = sc.nextInt();
            ArrayList<Double> list = new ArrayList<>();
            
            for (int j = 0; j < people; j++) {
                int logs = sc.nextInt();
                double total = 0;

                for (int k = 0; k < logs; k++) {
                    total += sc.nextDouble();
                }

                list.add(total);
            }

            Collections.sort(list);

            double curr = 0;
            double total = 0;
            for (int j = 0; j < list.size(); j++) {
                curr += list.get(j);
                total += curr;
            }

            System.out.println(total / (double) people);
        }
    }
}

File: Difficulty 3.0/Wood Cutting/WoodCutting.py
test = int(input())

# Settling each test case
for x in range(test):
    people = int(input())
    arrList = []

    # Adding all the people into the list
    for y in range(people):
        line = input().split(" ")
        logs = int(line[0])
        total = 0
        
        # Counting the total of logs for each person
        for z in range(1, logs + 1):
            total += int(line[z])
        arrList.append(total)

    arrList.sort()

    result = 0

    for num in arrList:
        result *= 2
        result += num

    print(str(result / people))
